source/plot/Axis.ooc

```diff
@@ -0,0 +1,176 @@
+				tickValue += this tick
+				position y += - scaling y * tick
+			}
+		}
+		result = result & "</g>\n" clone()
+		result
+	}
+
+	getFirstTickValue: func -> Float {
+		result: Float
+		if ((this min < this tick && this min > 0.0f) || (this min > this tick && this min > 0.0f && this min < 1.0f) )
+			result = this min + (this tick - Float modulo(this min, this tick))
+		else
+			result = this min - Float modulo(this min, this tick)
+		result
+	}
+
+	getTickSVG: func (tickValue, base: Float, position, numberOffset, tickMarkerOnOtherSideOffset, tickMarkerEndOffset: FloatPoint2D, textAnchor: String) -> String {
+		result := "<g desc='" clone() & tickValue toString() & "'>\n" clone()
+		if (this gridOn)
+			result = result & Shapes line(position, position + tickMarkerOnOtherSideOffset, "grey", FloatPoint2D new(5,5))
+		result = result & Shapes line(position, position + tickMarkerEndOffset, "black")
+		result = result & Shapes line(position + tickMarkerOnOtherSideOffset, position + tickMarkerOnOtherSideOffset - tickMarkerEndOffset, "black")
+		tickValue = base >= pow(10, this precision - 1) || base <= pow(10, - this precision) ? (tickValue / base) : tickValue
+		tempTick := tickValue toString()
+		tempTickInt := tickValue as Int toString()
+		result = result & Shapes text(position + numberOffset, tickValue == floor(tickValue) ? tempTickInt : tempTick, this fontSize, textAnchor)
+		tempTick free()
+		tempTickInt free()
+		result = result & "</g>\n" clone()
+		result
+	}
+
+	roundEndpoints: func {
+		if (this roundAxisEndpoints) {
+			this min = this roundToValueDigits(this min, 2, false)
+			this max = this roundToValueDigits(this max, 2, true)
+		}
+	}
+
+	roundToValueDigits: func (value: Float, valueDigits: Int, up: Bool) -> Float {
+		valueAndPower := this decomposeToValueAndPower(value, valueDigits)
+		result := valueAndPower[0]
+		if (valueAndPower[0] != 0) {
+			result = up ? ceil(valueAndPower[0]) : floor(valueAndPower[0])
+			result *= valueAndPower[1]
+		}
+		gc_free(valueAndPower data)
+		result
+	}
+
+	decomposeToValueAndPower: func (value: Float, valueDigits: Int) -> Float[] {
+		power := 1.0f
+		if (value != 0) {
+			while (Float absolute(value) > pow(10, valueDigits)) {
+				value /= 10.0f
+				power *= 10.0f
+			}
+			while (Float absolute(value) < pow(10, valueDigits-1)) {
+				value *= 10.0f
+				power /= 10.0f
+			}
+		}
+		result := Float[2] new()
+		result[0] = value
+		result[1] = power
+		result
+	}
+
+	getBase: func (value: Float, valueDigits: Int) -> Float {
+		valueAndPower := this decomposeToValueAndPower(value, valueDigits)
+		result := valueAndPower[1]
+		gc_free(valueAndPower data)
+		result
+	}
+
+	getScientificPower: func (value: Float) -> String {
+		// TODO: handle zero value
+		power := 1
+		while (value > 10) {
+			power += 1
+			value /= 10
+		}
+		while (value <= 1) {
+			power -= 1
+			value *= 10
+		}
+		result := "10^" clone() & power toString()
+		result
+	}
```

Comment:
Same here, these last 4 functions seem generic to Floats and shouldn't be specific to the Plot library.